rmf_task/test/integration/test_backups.cpp

```diff
@@ -15,24 +15,64 @@
  *
 */
 
+#include <iostream>
+#include <filesystem>
 #include <rmf_task/BackupFileManager.hpp>
 
 #include "../mock/MockDelivery.hpp"
 
 #include <rmf_utils/catch.hpp>
 
+const std::string backup_root_dir = "/tmp/rmf_task/test_backups";
+void cleanup() {
+  std::filesystem::remove_all(std::filesystem::weakly_canonical(backup_root_dir));
+}
+
+SCENARIO("Folder creation tests")
+{
+  cleanup();
+
+  rmf_task::BackupFileManager backup(backup_root_dir);
+  backup.clear_on_startup(true);
+  backup.clear_on_shutdown(true);
+
+  auto group_backup = backup.make_group("group");
+  CHECK(std::filesystem::exists(backup_root_dir + "/group"));
+
+  CHECK(!std::filesystem::exists(backup_root_dir + "/group" + "/robot"));
+  group_backup->make_robot("robot"); // Due to RAII and clear_on_shutdown the destructor is called immediately
+  CHECK(!std::filesystem::exists(backup_root_dir + "/group" + "/robot"));
+
+  auto robot_backup = group_backup->make_robot("robot"); 
+  CHECK(std::filesystem::exists(backup_root_dir + "/group" + "/robot"));
+
+  group_backup->make_robot("robot"); // Should not delete backup
+  CHECK(std::filesystem::exists(backup_root_dir + "/group" + "/robot"));
```

Comment:
Just like `robot_backup` you could test that the directory is removed after `group_backup` is cleared.

Reply:
6298276 and 0b66d33 adds this, surprisingly, this test fails. the group directory is not removed. Where can i find the implementation of `reset`?